common/src/main/java/com/wynntils/features/user/ItemTextOverlayFeature.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.mc.event.HotbarSlotRenderEvent;
+import com.wynntils.mc.event.SlotRenderEvent;
+import com.wynntils.mc.render.FontRenderer;
+import com.wynntils.wc.custom.item.WynnItemStack;
+import com.wynntils.wc.custom.item.properties.ItemProperty;
+import com.wynntils.wc.custom.item.properties.type.TextOverlayProperty;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@FeatureInfo(category = "Inventory")
+public class ItemTextOverlayFeature extends UserFeature {
+
+    @Config
+    public static boolean powderTierEnabled = true;
+
+    @Config
+    public static boolean powderTierRomanNumerals = true;
+
+    @Config
+    public static FontRenderer.TextShadow powderTierShadow = FontRenderer.TextShadow.OUTLINE;
+
+    @Config
+    public static boolean inventoryTextOverlayEnabled = true;
+
+    @Config
+    public static boolean hotbarTextOverlayEnabled = true;
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Post e) {
+        if (!inventoryTextOverlayEnabled) return;
+
+        drawTextOverlay(e.getSlot().getItem(), e.getSlot().x, e.getSlot().y, false);
+    }
+
+    @SubscribeEvent
+    public void onRenderHotbarSlot(HotbarSlotRenderEvent.Post e) {
+        if (!hotbarTextOverlayEnabled) return;
+
+        drawTextOverlay(e.getStack(), e.getX(), e.getY(), true);
+    }
+
+    public void drawTextOverlay(ItemStack item, int slotX, int slotY, boolean hotbar) {
+        if (!(item instanceof WynnItemStack wynnItem)) return;
+        if (!wynnItem.hasProperty(ItemProperty.TEXT_OVERLAY)) return;
+
+        for (TextOverlayProperty overlayProperty : wynnItem.getProperties(ItemProperty.TEXT_OVERLAY)) {
+            boolean contextEnabled = hotbar ? overlayProperty.isHotbarText() : overlayProperty.isInventoryText();
+            if (!overlayProperty.isTextOverlayEnabled() || !contextEnabled) continue; // not enabled or wrong context
+
+            TextOverlayProperty.TextOverlay textOverlay = overlayProperty.getTextOverlay();
+
+            PoseStack poseStack = new PoseStack();
+            poseStack.translate(0, 0, 300); // items are drawn at z300, so text has to be as well
```

Comment:
We should not draw it on a z on top of the item? Like 301 (or 299, if z goes the other way). It might not make any practical difference, but it seems more robust.

Reply:
We do Z 400 for the item highlights iirc

Reply:
So maybe Z 200 for text then?

Reply:
I thought highlights were placed behind the items, while text should be placed in front of. So I assumed that lower Z were closer to the viewer, and higher further away. But maybe my mental model is incorrect?

Reply:
Oh yeah so 300 should be fine then. If it's a problem we can fix that later.